pkg/controller/operators/catalog/operator.go

```diff
@@ -903,6 +915,11 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 		return err
 	}
 
+	failForwardEnabled, err := o.isFailForwardEnabled(namespace)
+	if err != nil {
+		o.logger.WithError(err).Debug("err")
```

Comment:
Are there any side effect letting the reconciliation go through even through we cannot tell whether we should failForward? Or will it just keep retrying until we can finally query the OG? If so, would it make sense to just log the error in the `isFailForwardEnabled` function and not return it?

Reply:
The only side effect that I ran into locally was that we were using a nil/unregistered OG lister which I believe resulted in an error being produced, and zero OG being able to be listed in a namespace. This seems like a potential rough edge, as there's potential to silently ignore the default behavior, or silently regress when attempting to list OGs in a namespace that contain this fail forward upgrade strategy as we'll treat an error there as a namespace using the default behavior.

Reply:
Great points Nick, I'm happy pursuing that implementation.